File: backend/src/main/java/com/awais/hr/module/expense/service/ExpenseService.java
package com.awais.hr.module.expense.service;

import com.awais.hr.module.expense.dto.ExpenseClaimRequestDTO;
import java.util.List;
import java.util.Map;

public interface ExpenseService {
    List<Map<String, Object>> getExpenses(String email);
    void submitExpense(String email, ExpenseClaimRequestDTO dto);
    void deleteExpense(String id);
    void uploadReceipt(String expenseId, String receiptUrl);
    void approveExpense(String expenseId, String email);
    void rejectExpense(String expenseId, String email);
}

File: frontend/src/services/expenseService.js
import apiClient from './api';

export async function getExpenses() {
  return apiClient.get('/suite/expense/claims');
}

export async function submitExpense(claimData) {
  return apiClient.post('/suite/expense/claims', claimData);
}

export async function approveExpense(id) {
  return apiClient.post(`/suite/expense/claims/${id}/approve`, {});
}

export async function rejectExpense(id) {
  return apiClient.post(`/suite/expense/claims/${id}/reject`, {});
}

export async function deleteExpense(id) {
  return apiClient.delete(`/suite/expense/claims/${id}`);
}
